fastai/callback/tensorboard.py

```diff
@@ -42,8 +43,98 @@ def after_epoch(self):
             x,y,its,outs = self.dls.valid.show_results(b, out, show=False, max_n=self.n_preds)
             tensorboard_log(x, y, its, outs, self.writer, self.train_iter)
 
+    def before_validate(self):
+        if self.projector:
+            self.run_projector = True
+
+            self.h = hook_output(self.learn.model[1][1] if not self.layer else self.layer)
+            self.feat = {}
+
+    def after_pred(self):
+        if self.run_projector:
+            self.feat = add_projector_features(self.learn, self.h, self.feat)
+
+    def after_validate(self):
+        if self.run_projector:
+            self.run_projector = False
+            self._remove()
+            write_projector_embedding(self.learn, self.writer, self.feat)
+
     def after_fit(self): self.writer.close()
 
+    def _remove(self):
+        if getattr(self, 'h', None): self.h.remove()
+
+    def __del__(self): self._remove()
+
+# Cell
+class TensorBoardProjectorCallback(Callback):
+    "Saves Embeddings for Tensorboard Projector"
+    def __init__(self, log_dir=None, layer=None):
+        store_attr('log_dir,layer')
+        self.run_projector = False
+
+    def before_validate(self):
+        self.run_projector = True
+
+        self.writer = SummaryWriter(log_dir=self.log_dir)
+        self.h = hook_output(self.learn.model[1][1] if not self.layer else self.layer)
+        self.feat = {}
+
+
+    def after_pred(self):
+        if self.run_projector:
+            self.feat = add_projector_features(self.learn, self.h, self.feat)
+
+    def after_validate(self):
+        if self.run_projector:
+            self.run_projector = False
+
+            self._remove()
+
+            write_projector_embedding(self.learn, self.writer, self.feat)
+
+
+    def _remove(self):
+        if getattr(self, 'h', None): self.h.remove()
+
+    def __del__(self): self._remove()
+
+# Cell
```

Comment:
Do this and the next one need to be public? If not, rename them to start with `_`